sagemaker-builder/packing-a-tensorflow-model/build_sagemaker_container.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import absolute_import
+
+import base64
+import contextlib
+import os
+import shlex
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import boto3
+
+IMAGE_TEMPLATE = "{account}.dkr.ecr.{region}.amazonaws.com/{image_name}:{version}"
+
+DOCKERFILE_TEMPLATE = """
+FROM {0}
+
+RUN apt-get update && apt-get install -y --no-install-recommends git
+
+RUN git clone https://github.com/mvsusp/sagemaker-containers.git \
+-b mvs-sagemaker-containers-train-improvements && cd sagemaker-containers \
+&& pip install . --quiet --disable-pip-version-check
+
+COPY {1} /opt/ml/code
+
+ENV PYTHONPATH /opt/ml/code:$PYTHONPATH
+ENV SAGEMAKER_TRAINING_MODULE {2}
+
+{3}"""
+
+
+def build(base_image, entrypoint, source_dir, tag, build_commands=None):
+    """
+    Build your algo using from a Docker `base_image`.
+
+    Args:
+        base_image (string): Docker image which your algo is based from.
+        entrypoint (string): Path (relative) to the Python source file which should be executed
+                as the entry point to training. This should be compatible with either Python 2.7 or Python 3.5.
+        source_dir (str): Path (absolute or relative) to a directory with any other training
+                source code dependencies aside from tne entry point file (default: None). Structure within this
+                directory are preserved when training on Amazon SageMaker.
+        tag (string): tag name
+        build_commands (string): additional instructions to send to the container
```

Comment:
should build_commands be a list? 

we need to make it clearer what user input should be. options could be 
- one string that is "CMD1 && CMD2 && ..."
- list of strings that we && together in one run command
- list of strings that each get put in own RUN command (and may or may not have && in the string)

# 3 seems like the most straight forward